pkg/cli/clisqlshell/editor_copy.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package clisqlshell
+
+import (
+	"os"
+
+	"github.com/cockroachdb/errors"
+)
+
+// copyEditor redirects the input to either a "main" editor
+// (outside of COPY mode) or a "copy" editor (inside COPY).
+// This is because the main editor may be multi-line and
+// thus inadequate for input of COPY data.
+type copyEditor struct {
```

Comment:
i'm not sure about the naming of this - it sounds like it handles copy but if i read this right it's really a mux of some sort where we redirect traffic based on the mode we're in. can we do something like `editorRouter` or something? file name has a similar issue.

Reply:
I went for `bimodalEditor`. Do you like it?